### PR TITLE
[NEW stable/busybox] added busybox chart

### PR DESCRIPTION
**Pull request by jfelten**
We use this to exercise our CI pipeline

### CODE REVIEW

**Comment by lachie83** [comment]
@k8s-bot ok to test

File: stable/busybox/values.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  type: ClusterIP
+  externalPort: 80
+  internalPort: 80
+resources:
+  limits:
+    cpu: 100m
+    memory: 128Mi
+  requests:
+    cpu: 100m
+    memory: 128Mi
```
Comment (seanknox) [inline stable/busybox/values.yaml:20]: These resources don't appear to be used. [Here's](https://github.com/kubernetes/charts/blob/master/stable/datadog/templates/daemonset.yaml#L20) an example of a chart using them.

**Comment by seanknox** [comment]
Works like a charm! :) One comment around resource quotas, but otherwise LGTM.

**Comment by jfelten** [comment]
ok I added a resources section per the review comments

File: stable/busybox/templates/_helpers.tpl [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+{{- default .Chart.Name .Values.nameOverride | trunc 24 | trimSuffix "-" -}}
+{{- end -}}
+
+{{/*
+Create a default fully qualified app name.
+We truncate at 24 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+*/}}
+{{- define "fullname" -}}
+{{- $name := default .Chart.Name .Values.nameOverride -}}
+{{- printf "%s-%s" .Release.Name $name | trunc 24 | trimSuffix "-" -}}
```
Comment (unguiculus) [inline stable/busybox/templates/_helpers.tpl:15]: I think you should truncate at 63 characters.

File: stable/busybox/values.yaml (hunk 1 above)
Comment (unguiculus) [reply stable/busybox/values.yaml:20]: I think it is best practice not to set resources by default. Here's an example: https://github.com/kubernetes/charts/blob/master/stable/nginx-ingress/values.yaml#L41

File: stable/busybox/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ template "fullname" . }}
+  labels:
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
```
Comment (unguiculus) [inline stable/busybox/templates/deployment.yaml:6]: Use best practices for labels.
https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/labels.md#standard-labels

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by seanknox** [comment]
Good suggestions @unguiculus. @jfelten would you mind updating based on @unguiculus's feedback?

File: stable/busybox/templates/deployment.yaml (hunk 3 above)
Comment (seanknox) [reply stable/busybox/templates/deployment.yaml:6]: 👍  to standard labels

re: not setting resources by default—while we don't call for this explicitly in our technical requirements, I can see the merit. Wouldn't consider it blocking for this PR.

**Comment by prydonius** [comment]
I'm not convinced this chart provides much value to the stable chart repo, which is intended for pre-packaging useful applications. This seems more like a debugging tool. Why would I use this instead of simply using `kubectl run`? What if I wanted to use alpine, or debian -- I don't see us having a chart for every base OS?

**Comment by seanknox** [comment]
@jfelten Agree with @prydonius on this. Charts are meant to facilitate services running in a Kubernetes cluster, and busybox as a one-off command doesn't fit that model. I'm sorry to we left you hanging as long as we did on this and will try to do a better job at assessing in the future.

**Comment by jfelten** [comment]
Sorry for not replying - I forgot about this.  We have an actual need for this in a client helm build pipeline, and from experience it is useful to spin up a pod through helm vs kubectl.  I do understand the need to keep the charts repo clean and purposeful and will publish this elsewhere.

**Pull request closed without merging**